async-nats/src/jetstream/consumer/push.rs

```diff
@@ -468,11 +470,14 @@ impl<'a> futures::Stream for Ordered<'a> {
                                         };
 
                                         let info = jetstream_message.info()?;
-                                        if info.stream_sequence != self.stream_sequence + 1 {
+                                        if info.consumer_sequence != self.consumer_sequence + 1
```

Comment:
We just do stream sequence in sync nats, wanna give a bit more rationale?

Reply:
If discard policy kicks in, you can't rely on stream sequence, so we have a secondary measure to check if we are really out of sync, or this is just a stream with discarded/deleted messages.

Reply:
Stream sequence is not guaranteed. Take an example: discard policy kicks in. We should rely on consumer sequence and if that is mismatched (because it was recreated for example), fallback to stream sequence.

Reply:
Hmm, we really are in need tests to check our assumptions.